bpf/lib/ipv4.h

```diff
@@ -22,6 +22,29 @@
 
 #include "dbg.h"
 
+#if defined IPV4_FRAGMENTS && defined HAVE_LRU_MAP_TYPE
+struct ipv4_frag_id {
+	__be32	daddr;
+	__be32	saddr;
+	__be16	id;		/* L4 datagram identifier */
+	__u8	nexthdr;
+} __attribute__((packed));
+
+struct ipv4_frag_l4ports {
+	__be16	dport;
+	__be16	sport;
+} __attribute__((packed));
+
+struct bpf_elf_map __section_maps IPV4_FRAG_DATAGRAMS_MAP = {
+	.type           = BPF_MAP_TYPE_LRU_HASH,
+	.size_key	= sizeof(struct ipv4_frag_id),
+	.size_value	= sizeof(struct ipv4_frag_l4ports),
+	.pinning	= PIN_GLOBAL_NS,
+	.max_elem	= CILIUM_IPV4_FRAG_MAP_MAX_ENTRIES,
+	.flags		= CONDITIONAL_PREALLOC,
+};
```

Comment:
If #10241 is merged first, you'll need to open this map from the cilium-agent, in `daemon.InitMaps()`. (If not, I can do it when I rebase.) To do that, you can have a look at ab4b45623cce7a77 ("maps, daemon: Create all global maps from daemon") and add similar code as I did for e.g. the `signalmap`. Note that you'll need to update the Makefile to generate the deepcopy file. There's already code in `daemon.InitMaps()` to detect if LRU hashmaps are supported.

Reply:
I'll definitely look at that and try to open it from cilium-agent directly, let's try to avoid a rebase :). Thanks for the heads up and pointers!